osquery/tables/system/darwin/processes.cpp

```diff
@@ -266,6 +268,66 @@ void genProcNumThreads(QueryContext& context, int pid, Row& r) {
   }
 }
 
+uint64_t parseKCItem(kcdata_item_t item) {
+  if (item->type == STACKSHOT_KCTYPE_TASK_SNAPSHOT) {
+    struct task_snapshot_v2* tsv2 = (struct task_snapshot_v2*)item->data;
+    return tsv2->ts_unique_pid;
+  } else {
+    return ULLONG_MAX;
+  }
+}
+
+void genProcUniquePid(QueryContext& context, int pid, Row& r) {
+  if (!context.isColumnUsed("unique_pid")) {
+    return;
+  }
+
+  // A definition of the stackshot_config structure so we can pull data out of
+  // the kernel
+  typedef struct stackshot_config {
+    int pid;
+    uint32_t flags;
+    uint64_t dtimestamp;
+    uint64_t buf;
+    uint32_t buf_size;
+    uint64_t out_buf_addr;
+    uint64_t out_buf_size_addr;
+  } stackshot_config_t;
+
+  char* addr = nullptr;
+  uint64_t size = 0;
+
+  stackshot_config_t stconf = {0};
+  stconf.pid = pid;
+  stconf.buf = 0;
+  stconf.buf_size = 0;
+  stconf.flags = 0x10000; // STACKSHOT_KCDATA_FORMAT
+  stconf.out_buf_addr = (uint64_t)&addr;
+  stconf.out_buf_size_addr = (uint64_t)&size;
+
+  auto rc = syscall(491, 1, (uint64_t)&stconf, sizeof(stconf));
+
+  if (rc < 0) {
+    r["unique_pid"] = INTEGER(-1);
+    return;
+  }
+
+  uint64_t off = 16;
```

Comment:
Just a question, what does number 16 mean here?
And is it the same number 16 in the code below here?
Would you like to create a constant for it?